core/src/com/unciv/logic/github/KtorGithubAPI.kt

```diff
@@ -0,0 +1,120 @@
+package com.unciv.logic.github
+
+import com.unciv.UncivGame
+import io.ktor.client.*
+import io.ktor.client.engine.cio.*
+import io.ktor.client.plugins.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.http.*
+import kotlinx.coroutines.delay
+import kotlin.time.Clock
+import kotlin.time.ExperimentalTime
+import kotlin.time.Instant
+
+object KtorGithubAPI {
+    /**
+     * @see <a href="https://ktor.io/docs/client-default-request.html#url">
+     *          Ktor Client > Developing applications > Requests > Default request > Base URL
+     *      </a>
+     */
+    const val baseUrl = "https://api.github.com"
+
+    /**
+     * Add a bearer token here if needed
+     *
+     * @see <a href="https://github.com/yairm210/Unciv/issues/13951#issuecomment-3326406877">#13951 (comment)</a>
+     */
+    const val bearerToken = ""
+
+    private val client = HttpClient(CIO) {
+        followRedirects = true
+        install(HttpRequestRetry) {
+            maxRetries = 3
+            retryOnException()
+        }
+        defaultRequest {
+            url(baseUrl)
+            header("X-GitHub-Api-Version", "2022-11-28")
+            header(HttpHeaders.Accept, "application/vnd.github+json")
+            userAgent(UncivGame.getUserAgent("Github"))
+            if (bearerToken.isNotBlank()) bearerAuth(bearerToken)
+        }
+    }
+
+    /**
+     * Wait for rate limit to end if any and returns true if there was any rate limit
+     */
+    @OptIn(ExperimentalTime::class)
+    private suspend fun consumeRateLimit(resp: HttpResponse): Boolean {
```

Comment:
Intentional. As someone have previously mentioned in `class RateLimit` ratelimits are not always global. and vary between routes.


also even with the previous "global" ratelimit, we are not checking for a rate limit first to begin with. we wait to get an error and then check rate limits.